# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/x86-shared/Lowering-x86-shared.h
# Commit: 1e582b17eab7
# Full Hash: 1e582b17eab7abf3958e46bc52ea3eeafa764eeb
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r a309060018a8 -r 1e582b17eab7 js/src/jit/x86-shared/Lowering-x86-shared.h
--- a/js/src/jit/x86-shared/Lowering-x86-shared.h	Wed Sep 09 06:37:59 2020 +0000
+++ b/js/src/jit/x86-shared/Lowering-x86-shared.h	Wed Sep 09 08:59:55 2020 +0000
@@ -46,6 +46,7 @@
   void lowerUMod(MMod* mod);
   void lowerUrshD(MUrsh* mir);
   void lowerPowOfTwoI(MPow* mir);
+  void lowerWasmBuiltinTruncateToInt32(MWasmBuiltinTruncateToInt32* ins);
   void lowerTruncateDToInt32(MTruncateToInt32* ins);
   void lowerTruncateFToInt32(MTruncateToInt32* ins);
   void lowerCompareExchangeTypedArrayElement(